### PR TITLE
Supporting transaction compression

### PR DESCRIPTION
**Pull request by bradlhart**
## Change Description
Adding the optional ability to use zlib compression (or other future compression libraries) for the transaction.

See https://github.com/EOSIO/eosjs/issues/551

## API Changes
- [x] API Changes
`compression` option for transaction methods

## Documentation Additions
- [x] Documentation Additions
Information about the compression ability and how to use it for a transaction

### CODE REVIEW

File: src/eosjs-api.ts [hunk 1]
```diff
@@ -260,13 +260,15 @@ export class Api {
...
      *    * `sign`: sign this transaction?
+     *    * `compression`: compress this transaction?
      *    * If both `blocksBehind` and `expireSeconds` are present,
      *      then fetch the block which is `blocksBehind` behind head block,
      *      use it as a reference for TAPoS, and expire the transaction `expireSeconds` after that block's time.
      * @returns node response if `broadcast`, `{signatures, serializedTransaction}` if `!broadcast`
      */
-    public async transact(transaction: any, { broadcast = true, sign = true, blocksBehind, expireSeconds }:
-        { broadcast?: boolean; sign?: boolean; blocksBehind?: number; expireSeconds?: number; } = {}): Promise<any> {
+    public async transact(transaction: any, { broadcast = true, sign = true, compression, blocksBehind, expireSeconds }:
```
Comment (jeffreyssmith2nd) [inline src/eosjs-api.ts]: It may be time to refactor this out into a `TransactParams` type, it's getting a bit large.

Comment (jeffreyssmith2nd) [reply src/eosjs-api.ts]: Or actually I think it would be `TransactArgs`, I think that's the pattern elsewhere.

Comment (bradlhart) [reply src/eosjs-api.ts]: I used TransactConfig in https://github.com/EOSIO/eosjs/pull/677

Comment (jeffreyssmith2nd) [reply src/eosjs-api.ts]: AH right, that works 👍

File: package.json [hunk 2]
```diff
@@ -40,6 +41,7 @@
     "@blockone/tslint-config-blockone": "3.0.0",
     "@types/jest": "24.9.1",
     "@types/node": "11.9.4",
+    "@types/pako": "^1.0.1",
     "cypress": "3.8.2",
     "eosjs-ecc": "^4.0.7",
```
Comment (jeffreyssmith2nd) [inline package.json]: Should `@types/pako` also be version locked?

And probably unrelated to this PR, but can we remove `eosjs-ecc` here?

File: src/eosjs-api.ts [hunk 3]
```diff
@@ -242,19 +245,32 @@ export class Api {
         };
     }
 
+    /** Deflate a serialized object */
+    public deflateSerializedArray(serializedArray: Uint8Array) {
```
Comment (jeffreyssmith2nd) [inline src/eosjs-api.ts]: Add the return type here

File: package.json (hunk 2 above)
Comment (bradlhart) [reply package.json]: It doesn't seem like devDependencies have usually been locked but I can update it.

`eosjs-ecc` is still there for the `eosjs-ecc-verification.test.js` file that confirms that `elliptic` and `eosjs-ecc` match, but I don't think we will need it on release.  Let me know if you'd like me to remove them in this PR or another one.

Comment (jeffreyssmith2nd) [reply package.json]: We can leave it in this PR for now then, and figure out with the team when is an appropriate time to remove it.

File: src/tests/web.html [hunk 4]
```diff
@@ -288,6 +290,7 @@ <h1>Web Build Integration Tests</h1>
             <div><h2>Transact with blocksBehind Configuration Parameter</h2><button onClick='testTransactWithConfigBlocksBehind(event);'>Test</button></div>
             <div><h2>Transact with useLastIrreversible Configuration Parameter</h2><button onClick='testTransactWithConfigUseLastIrreversible(event);'>Test</button></div>
             <div><h2>Transact with Manually Configured TAPOS</h2><button onClick='testTransactWithoutConfig(event);'>Test</button></div>
+            <div><h2>Transact with Manually Configured TAPOS</h2><button onClick='testTransactWithCompression(event);'>Test</button></div>
```
Comment (jeffreyssmith2nd) [inline src/tests/web.html]: Transact with Compression

**Pull request merged**